x-pack/plugins/security_solution/scripts/run_cypress/parallel.ts

```diff
@@ -101,26 +100,26 @@ export const cli = () => {
       );
 
       if (argv.changedSpecsOnly) {
-        const basePath = process.cwd().split('kibana/')[1];
-        files = findChangedFiles('main', false)
-          .filter(
-            minimatch.filter(path.join(basePath, cypressConfigFile?.e2e?.specPattern), {
-              matchBase: true,
-            })
-          )
-          .map((filePath: string) => filePath.replace(basePath, '.'));
-
-        if (!files?.length) {
-          // eslint-disable-next-line no-process-exit
-          return process.exit(0);
-        }
+        files = (findChangedFiles('main', false) as string[]).reduce((acc, itemPath) => {
+          const existing = files.find((grepFilePath) => grepFilePath.includes(itemPath));
+          if (existing) {
+            acc.push(existing);
+          }
+          return acc;
+        }, [] as string[]);
 
         // to avoid running too many tests, we limit the number of files to 3
         // we may extend this in the future
         files = files.slice(0, 3);
```

Comment:
Will it always pick up the same 3 on a grep if there are more than 3 changed files?

Reply:
correct, until we reach the point where all our tests will be `burn-ready` I have limited it to just 3 files to avoid tests timeout